File: 263 Ugly Number.java
class Solution {
    public boolean isUgly(int num) {
        if (num <= 0) return false;
        if (num == 1) return true; // 不用這行也能pass, 但實測效率比較好，可能測資有很多筆1?
        while((num & 1) == 0) num >>= 1;
        while(num % 3 == 0) num /= 3;
        while(num % 5 == 0) num /= 5;
        return (num == 1);
    }
}

File: 263 Ugly Number.py
class Solution:
    def isUgly(self, num: int) -> bool:
        if num <= 0: return False
        if num == 1: return True
        while not num & 1:
            num >>= 1
        while not num % 5:
            num //= 5
        while not num % 3:
            num //= 3
        return num == 1